src/main/java/uk/ac/ebi/ddi/ws/util/MapUtils.java

```diff
@@ -18,9 +18,12 @@ public static <K, V extends Comparable<? super V>> Map<K, V> sortByValue(Map<K,
 
     public static Map<String, String> eliminateSet(Map<String, Set<String>> input) {
         Map<String, String> result = new HashMap<>();
-        for (String key : input.keySet()) {
-            result.put(key, input.get(key).iterator().next());
+        input.forEach((key, value) -> {
+            if (input.get(key).iterator().hasNext()) {
```

Comment:
It would be clearer to use `if (input.get(key).size() > 0)` here, IMO - avoids creating the iterator twice.

Reply:
I think it should be functionally equivalent, but you could write a couple of unit tests to check. You could also use `if (!input.get(key).isEmpty())`, which is even shorter than checking the size.